homeassistant/components/august/binary_sensor.py

```diff
@@ -127,81 +115,35 @@ def is_on(self):
-        ]
-        lock_door_state = await async_state_provider(self._data, self._door)
-        self._available = (
-            lock_door_state is not None and lock_door_state != LockDoorStatus.UNKNOWN
-        )
-        self._state = lock_door_state == LockDoorStatus.OPEN
-
         door_activity = await self._data.async_get_latest_device_activity(
             self._door.device_id, ActivityType.DOOR_OPERATION
         )
+        detail = await self._data.async_get_lock_detail(self._door.device_id)
 
         if door_activity is not None:
-            self._sync_door_activity(door_activity)
-
-    def _update_door_state(self, door_state, update_start_time):
-        new_state = door_state == LockDoorStatus.OPEN
-        if self._state != new_state:
-            self._state = new_state
-            self._data.update_door_state(
-                self._door.device_id, door_state, update_start_time
-            )
-
-    def _sync_door_activity(self, door_activity):
-        """Check the activity for the latest door open/close activity (events).
+            update_lock_detail_from_activity(detail, door_activity)
 
-        We use this to determine the door state in between calls to the lock
-        api as we update it more frequently
-        """
-        last_door_state_update_time_utc = self._data.get_last_door_state_update_time_utc(
-            self._door.device_id
-        )
-        activity_end_time_utc = dt.as_utc(door_activity.activity_end_time)
+        lock_door_state = None
+        if detail is not None:
+            lock_door_state = detail.door_state
 
-        if activity_end_time_utc > last_door_state_update_time_utc:
-            _LOGGER.debug(
-                "The activity log has new events for %s: [action=%s] [activity_end_time_utc=%s] > [last_door_state_update_time_utc=%s]",
-                self.name,
-                door_activity.action,
-                activity_end_time_utc,
-                last_door_state_update_time_utc,
-            )
-            activity_start_time_utc = dt.as_utc(door_activity.activity_start_time)
-            if door_activity.action in ACTIVITY_ACTION_STATES:
-                self._update_door_state(
-                    ACTIVITY_ACTION_STATES[door_activity.action],
-                    activity_start_time_utc,
-                )
-            else:
-                _LOGGER.info(
-                    "Unhandled door activity action %s for %s",
-                    door_activity.action,
-                    self.name,
-                )
+        self._available = lock_door_state != LockDoorStatus.UNKNOWN
+        self._state = lock_door_state == LockDoorStatus.OPEN
 
     @property
     def unique_id(self) -> str:
         """Get the unique of the door open binary sensor."""
-        return "{:s}_{:s}".format(
-            self._door.device_id,
-            SENSOR_TYPES_DOOR[self._sensor_type][SENSOR_NAME].lower(),
-        )
+        return "{:s}_open".format(self._door.device_id)
```

Comment:
f-strings for the win :)

```suggestion
        return f"{self._door.device_id}_open"
```